script.js

```diff
@@ -1,4 +1,134 @@
+  }
+
+  console.log("SPS is " + numberToSPS);
+  return numberToSPS;
+};
+
+// Create main function where user inputs "scissors", "paper" or "stone" and receives a result based on the following conditions:
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var programmeOutputsSPS = assignNumberToSPS();
+  console.log("programe outputs " + programmeOutputsSPS);
+  //creating variable for messages
+  var reversedGame =
+    "Oh and btw - CONGRATS! You found the secret but perhaps not much more thrilling 'reversed' game.";
+  //if user enters invalid input
+  var myOutputValue =
+    "Did you enter 'scissors', 'paper' or 'stone' exactly? This game is case-sensitive and a spelling nazi! :p <br><br> Oh, and here's a tip: add 'reversed' and a space to any of the above options, and see what happens to the game!";
+
+  // Create conditions
+  //Win
+  // If input is Sc and output is P, or if input is P and output is St, or if input is St and output is Sc, user wins
+
+  if (
+    input !== reverseScissors &&
+    input !== reversePaper &&
+    input !== reverseStone
+  ) {
+    reversedGame = "";
+  }
+
+  if (
+    ((input == scissors || input == reverseStone) &&
+      programmeOutputsSPS == paper) ||
+    ((input == paper || input == reverseScissors) &&
+      programmeOutputsSPS == stone) ||
+    ((input == stone || input == reversePaper) &&
+      programmeOutputsSPS == scissors)
+  ) {
+    myOutputValue =
+      "You won! Cheap thrill, eh?  <br><br> You typed in " +
+      input +
+      " and the computer chose " +
+      programmeOutputsSPS +
+      ". <br><br>" +
+      reversedGame;
+    console.log("User wins");
+  }
+
+  // Draw
+  // if input is Sc and output is Sc, or if input is P and output is P, or if input is St and output is St, user ties
+
+  if (
+    input == programmeOutputsSPS ||
+    (input == reversePaper && programmeOutputsSPS == paper) ||
+    (input == reverseStone && programmeOutputsSPS == stone) ||
+    (input == reverseScissors && programmeOutputsSPS == scissors)
+  ) {
+    myOutputValue =
+      "No win no lose - you drew! <br><br> You typed in " +
+      input +
+      " and the computer also chose " +
+      programmeOutputsSPS +
+      ". <br><br>" +
+      reversedGame;
+    console.log("User draws");
+  }
+
+  // Lost
+  // If input is Sc and output is St, or if input is P and output is Sc, or if input is St and output is P, user loses
+
+  if (
+    ((input == scissors || input == reversePaper) &&
+      programmeOutputsSPS == stone) ||
+    ((input == paper || input == reverseStone) &&
+      programmeOutputsSPS == scissors) ||
+    ((input == stone || input == reverseScissors) &&
+      programmeOutputsSPS == paper)
+  ) {
+    myOutputValue =
+      "You lost to a computer, but eh, no biggie! <br><br> You typed in " +
+      input +
+      " and the computer chose " +
+      programmeOutputsSPS +
+      ". <br><br>" +
+      reversedGame;
+    console.log("User loses");
+  }
```

Comment:
maybe you can consider writing helper functions to help you determine this part and shifting this block of code out of the main function ?